### PR TITLE
Package maintenance and pytest workflow

### PR DESCRIPTION
**Pull request by Dens49**
- Changed minimum python version to 3.11, due to the features used in https://github.com/quality-match/hari-client/pull/1
- Added github actions workflow for running pytest tests when creating a PR towards `main`
- Link Changelog in Readme
- Explain development setup in Readme

### CODE REVIEW

File: .github/workflows/hari-client-pytest.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    branches:
+      - main
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        python-version: [3.11, 3.12]
```
Comment (katia-qm) [inline .github/workflows/hari-client-pytest.yaml:14]: How will this be maintained in the future with new versions?

**Review by katia-qm — APPROVED** [review]
Kewl! I only don't really get, why pre-commit-test workflow is necessary. As long as developers don't skip it, it will act exactly like the local one, right?

**Comment by Dens49** [comment]
> Kewl! I only don't really get, why pre-commit-test workflow is necessary. As long as developers don't skip it, it will act exactly like the local one, right?

IMO It's the same reasoning as for having automated tests in the PR: you should have them running locally, but the automated checks in the PR make it visible to the reviewer that everything passed.

File: .github/workflows/hari-client-pytest.yaml (hunk 1 above)
Comment (Dens49) [reply .github/workflows/hari-client-pytest.yaml:14]: When a new python version is released, we can just add it to the matrix. If tests for the new version fail, we know we have to do some changes before being able to support the new version. Right now, the setup.py file states that the package is working with python 3.11 and 3.12

**Pull request merged**